src/klein/_interfaces.py

```diff
@@ -7,18 +7,37 @@
 works, since mypy doesn't otherwise get along with Zope Interface.
 """
 
-from typing import TYPE_CHECKING
+from typing import Mapping, Optional, TYPE_CHECKING, Text
+
+from zope.interface import Attribute, Interface
 
-from ._iapp import IKleinRequest
 from ._imessage import (
     IHTTPHeaders as _IHTTPHeaders,
     IHTTPMessage as _IHTTPMessage,
     IHTTPRequest as _IHTTPRequest,
     IHTTPResponse as _IHTTPResponse,
     IMutableHTTPHeaders as _IMutableHTTPHeaders,
 )
-
-IKleinRequest  # Silence linter
+from ._typing import ifmethod
+
+
+class IKleinRequest(Interface):
+    branch_segments = Attribute("Segments consumed by a branch route.")
+    mapper = Attribute("L{werkzeug.routing.MapAdapter}")
+
+    @ifmethod
+    def url_for(
+        request,  # type: IKleinRequest
```

Comment:
If Klein's master branch is dropping support for Python 3.5, it might be worth using variable annotations here.
But that can be done in a separate PR.

Reply:
Yeah. I've got a lot of momentum pent up using 2.7 syntax, but plan to convert after I get the work in #301 (which this is a piece of) done.